source/Cosmos.HAL2/DeviceFS.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Cosmos.HAL2
+{
+    public class DeviceFS : StorageDevice
+    {
+        public char Seperator = '/';
+
+        public override void Chmod(string f, string perms)

```

Comment:
We can't use chmod, and we can't implement it because we're using FAT and FAT doesn't support chmod if I remember well.(but I'm not sure)

Reply:
I put it in there, because why not. even if we don't use it right away. I'll develop a work around if I have a few days.

Reply:
Okay "Technically" I cannot support chmod, but "Technically" we can just pop an if statement in file creation if an user can do it.